### ISSUE TITLE
Extend MPD protocol with spotify-specific commands

### ISSUE THREAD

**Issue by arybczak** [mopidy/mopidy-mpd#39]
Hey,

ncmpcpp author here. I've been recently looking into using ncmpcpp with mopidy-spotify so that I don't have to alternate between MPD and the official Spotify client.

However, it looks like it's not possible to have more specific control over Spotify with the MPD protocol, e.g. adding tracks/albums/artists to the collection.

I thought about it for a while and the potential solutions I see are:

1) Make it possible for ncmpcpp to use mopidy-http api instead of the MPD protocol
2) Extend mopidy-mpd with spotify-specific commands so that ncmpcpp (or other mopidy-compatible MPD clients) can support them conditionally after inspecting the output of `commands`.
3) Give up and fall back to MPD/Spotify.

(1) is a lot of work that I don't have time for and (3) isn't very satisfying, so I'd like to tackle (2) if the idea makes sense to you.

**Comment by kingosticks** [comment]
What sort of Spotify specific things are you wanting to do? We don't have any support for manipulating the user's My Music collection. Is that the "collection" you are referring to?

**Comment by arybczak** [comment]
> We don't have any support for manipulating the user's My Music collection. Is that the "collection" you are referring to?

Hmm. I looked at mopidy-iris and it can at least
* add tracks from queue to collection (but can't remove them)
* add albums from search to collection (can remove them)

Doesn't it do it through mopidy?

Looks like it can't add/remove artists though.

**Comment by kingosticks** [comment]
No, it doesn't go through Mopidy. Iris does some calls directly to Spotify's Web API. That's why you have to authorise Iris separately. 

Traditionally Mopidy-Spotify only supported things provided by libspotify (which is deprecated and pre-dates My Music, Spotify Connect , etc). We've been moving things over to the Web API where it makes sense. But Mopidy's Core API doesn't have anything that obviously maps to the Spotify-specific My Music collection. We try to avoid anything backend-specific. We might be able to support it through a special playlist, but we don't do that currently.

**Comment by Eluminae** [comment]
I just discovered it is impossible to search through playlists with mopidy-mpd (spotify playlists in this example).

Is it possible to add some mpd tag ("Playlist") so that requests like "mpv search Playlist foobar" would result in spotify uris ? Or is it something that Mopidy-Spotify should do ? Ideally I'd like not to use the Web extensions for this.

(Anyway it looks like clients like ncmpcpp would not be able to handle those extended tags but maybe the "any" tag would be enough for most use cases).

What are you thoughs about this ? Doable ? Recommendable ?
